### ISSUE TITLE
Add rule name to parse error messages

### ISSUE THREAD

**Issue by amiramix** [stylelint/stylelint#5144]
**Issue**

Stylelint reports CssSyntaxError when linting a statement that contains both, CSS variable and SASS interpolation variable.

**Reproduction**

The following Sass snippet causes Stylelint to incorrectly report an error, which then can't be suppressed. The only way to make Stylelint pass is to exclude the file from linting (which I would consider a critical bug in Stylelint).

Save as `test.scss`:

```scss
/* stylelint-disable */
$nav-height: 60px;

.hamburger-container {
  --page-height: 100px;
  margin-left: 40px;
  height: calc(var(--page-height) - 100vh + #{$nav-height});
  pointer-events: none;
}
```

Then lint with `npx stylelint test.scss`

The result is:

```shell
<input css 1>
 1:30  ✖  Unknown word   CssSyntaxError
```

Removing either `var(--page-height)` or `#{$nav-height}`, as well as replacing either of them with an absolute value, e.g. `60px`, will let Stylelint to parse the file properly. Looks like the issue is only when both types of variables are used in the same statement.

**Configuration**

```
const path = require('path');

module.exports = {
  extends: 'stylelint-config-recommended-scss',
  plugins: [
    'stylelint-scss',
    'stylelint-value-no-unknown-custom-properties'
  ],
  ignoreFiles: [
    '**/*.js',
    '**/dist/**',
    '**/build/**'
  ],
  rules: {
    'no-descending-specificity': null,
    'at-rule-no-unknown': null,
    'scss/at-rule-no-unknown': true,
    'csstools/value-no-unknown-custom-properties': [true, {
      importFrom: [
        path.resolve(__dirname, 'packages/rcom-raptor-ui_atomic/styles/_variables.css')
      ]
    }]
  }
}
```

**Version**

`13.10.0`

**Command**

`npx stylelint test.scss --config .stylelintrc.js`

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

"Yes, it's related to SCSS variables interpolation."

> What did you expect to happen?

"No error, or at the very least, the ability to suppress the error."

> What actually happened (e.g. what warnings or errors did you get)?

```shell
<input css 1>
 1:30  ✖  Unknown word   CssSyntaxError
```

**Comment by jeddy3** [comment]
@amiramix Thanks for the detailed report.

I believe stylelint and the built-in rules [can parse it](https://stylelint.io/demo#N4Igxg9gJgpiBcID0AqABAZwC4E8A2MeAlgHZYC0URGAhgEYFopIA6JAJCTQG7kAWMIgHM+WeGgBsABgAOADwDcbNgDo+NALZ0ArgCchMXeUhkapQ2mBs0acuRk0D-QSLFoAjFNmLraDTX1ScgIAMzcAFm8lEhsBYVFxMBo8MAAKbgDUuwcnONcASlsPL24+NABqNABiYE4eZ3isAF986JsZCFIsQ3IYbhgyDHESCBIYaKaQABoQDBxTOQRZsAwMafBRkOElqxi0FhBdbQIMA-FdmxsDmgojgnIR8m0SAGsRgHcSM7QsI5gp3xXEAMCBgF4PCC9DQyXDfX7af6A-YbPAQIyPUgZYhQZxyOF-AF7IGQDQaAYUR4waGwhA-AlIg6wMB4AI3IijcggsEQyjaGTEJLdey6CAyQxYIgwU60gDaSJs8MRRMuF0uaoOwhGuhg3zlyrVxNGGBgYG0Ev6vP5REFUvI7yIWD4lCIIRChnJ5CxCOl8suAF1fU0kX7CeqQEyWbo2RyueDHhg+GjHTQSDiZCKxbpcOQIP1dLoiLBpeJFaHLgcQqMKCFNEQ8DgeVA+QKbjAHpopfiEWWgZWyOQaxo6w3HkPVqQhOQDGMC2AB7X6+QXjAcO80VAu0qwyFnmAJRykikeZjkoXNz3kTuSHv2SRjMk549nhgHGAYDjRYYbmjzwyQFebw5YgxgCKcoyoD1HhGEhsBTKAAhxKhtUAr5aVLP9lxwEIozJSgTUjaM7wxaEkxTLBf2VA4ySoGgBxgG49DbLhcKfV4PlQkt6UokBHiLN9UwncgXxNF1rQdHAKLDXjm2tVtyBucgiBIrNyDuTs0K4qTISbK0bSEwgTSwNFizpbs-0pakGwwCA9DfSTyx4yEYDkX5aONIdIFRGD7KBDESCxQtKBshg2wwFkE2MCBSXJEz0O49NPyzEdIWeN4IE+HzkWNAg9zRexjW0KBIWZGhVh5VL2Myg5ssMvKZAKoregIMl+1YtKMo0szuJq3KjFwMVyrY9KONMrcHOwAsSEnKCYHeYCdU6sagWeB1Bvaka4rDBS1KhGEG3mzkYErbVdV9a48HeGgcGlC8bFVA1kWct8YVO-UDQOWMTgweSwh6WgWI7TlUTBL6Dlu97-yIXRsAeKVug3EBfRsEMkeRTU0QW8Q9QerbfqMEkWvIxG3uRwNgwva5bmOJjAaSY1vgOVF3kMMG-226n21wl8aDfH7ul0BmQGSS7rqEid7nm1nuPZ+53OtCAvNhubzD5lnaXOkWbr-WNjFRDBBLoKNeapPbghVugjoxwWxjzKWwx15kIH1qbOSNpjZoOmg8cF4Wrq17iHb1g23aVg6LeOzHkV90WNGOCUzbGO2HMDp3g550LXzbcOrfVoWLr9sWpol8wk6BHXPxINPebGZWxlVgXc+j77Y7wePJembXgfBCuq4z9P659-PRedoRi8TjuA67nMxUrl3Df77neezk7G6H-2w08vKBDke96dzpm1YplEt+c4IBiER1BYTJNS+RAmPRN7Mw8tlesbOvPNdvu7UYOJ6YBe2UqNexQxhmMbA74DioxRiTNGQgtSR2xjjIE2B8CEC6JFUkcEfQwIDG9IMypoEbyioTO0fAHRSkzopGChZI4az9rfA4ppsBRWFIlbMj99rmxfgg9+Tcv6WB-iAP+AC34wKBF7fmxhtDMI0KwzMNJwbbhARSOGEDiZIMITjDUcCc6iKQeIvGGDCb8LDKQfWsBC6jzbGHLukCcFkwIUfCMrJ9x3joCmSci82wSLVuIA4NtD5-mcVGVxnIPFCUocvWhH96ETzDMEwiQNQTgjlpvIiHsVY+Ibn4mJMc45EATgtJx+EXG3iSdyVJCsOReIHqvTWlix5FKCSUkJZSdaVMVjUqJ1s+iBO4gk0J7TxbWJVv4HeAzbwmXcMUkqrSYxT1cnWQSHTRiD0-nEhyEyDxVPSbXbx3s6kFxbm3EuGygRbLvGkiJ-cslrILiPRpDDwwtMSVcrp3Cem2zOciC5u0n5cIjq9B6dDrr8PuhDYRRM9H6OuIYi5JiHJbGhio8BCMoGKNgfAoFMKhaGPvmQBFQIzE0IaSMuusY7EaIcWqTRvZdyhIJrRGunsDk5KbuQY5BT25HwAgy4hblKG3MOcPYZhSnm8rKYy65S8Pm5wCQ3Hl9KyljL+ZwsB3wpCKuvKE5ibY6bRIPgqv8EqOQOG1GQAQxpvrMpVsS2Ady8mty5acrVKF7ABHJJa20NS7XRPlaSsV3yKxKo5O8Mh3QalCrZWvJ5pBYCmFcd8AATEfAgU1HTkAAF6GEhKxB0VUQAqo4YUqZR9qJEFom6Bi2pIqdMFayqOMag2FvfBWui1a9U7OlVnWVOT5VPPLZW+iWBGKcz1aVA16U+lhkHe2kdNazWeqlN6yhvrPnTocrOqt862xRimm2RK34jCRobSC9em7W1Do7apDxB7MxHu7YdQFwrz1AlnQARwRLoThMMpU2rrlGxt9TOWBrLZe8gn7DA-ooFKk9-MHXfQeWSpp3EP1fug0YgV-dulyt6Ua7ilIFakPIV4gto5qAj16KmHMIQhI2V0NXDJ49OJdSktoLQPQCA0CoC7bNIoENPJIOxi26JISLOApOPjTsC0JXkQ2fVgtDVPJ2sW5+z7oVbTXhy-JoHfTgrDJC7FSCKzKNhqiylONaUQ3Rq-NAiDjO4skfiomUDqX+iPj1IyRgbi-CIDoIU88wQwCwIh1d1D7W4a+R5gyvV5JYF8-5u9X4vOPsA-4vDTzPN5R8wWRL09kt5Xeep5E-bm1ZfxlFOgpAH1wd8UB2J0WcopZJFVrgKWiu6Pq6CsrMWUt8QGPBfsLXqspagtUzOBbytm1-fy0OmTT25NfVlXreViAzcwY+nDfaMs9aa3lItlkS0asa7VIw9UYCFWKuFb6Cn95Tvw2GKb53Lu61Kt9RdFrl2hf7muyLG7kErbOw1JyzUPS3ZyUp3bp38oXcagZEhVyRhYEIoLIq2gQqZcB+Qfq4694Q-u08laMGJ2KYJ82r0bZMJrl0DicHyJIdHwp9NmDs3-37Pgy+7TTrdPcSZ2tlnG3avZK64h0V3K-x8+oAL-wm3e0lZ24z5ICJmccpoDvVT5gTJSF8PgyYTQgA). It's likely the issue with one of the 3rd party plugins.

It could be `csstools/value-no-unknown-custom-properties` as it uses [postcss-value-parser under the hood](https://github.com/csstools/stylelint-value-no-unknown-custom-properties/blob/98f85f6fef1cf811f2f362f4a9a4622c5951f62b/src/lib/validate-decl.js#L1-L11).

The value parser [can be configured to ignore unknown words and interpolation](https://github.com/shellscape/postcss-values-parser/blob/master/docs/Exports.md#ignoreunknownwords). However, the authors of the plugin may have chosen not to do that as their plugins seem geared towards standard (future-facing) CSS syntax.

I suggest identifying which plugin is causing the error and then raising an issue in their respective issue tracker. They may be able to help out.

---

It would a better user experience in stylelint if it signalled which rule or plugin threw the error.  I'll label the issue as a discussion for that.

**Comment by amiramix** [comment]
Thank you @jeddy3 , it makes sense. Next week I will try to identify the plugin and report my findings. But I totally agree, if Stylelint could report the actual plugin or rule that caused this issue that would be a much better user experience.

**Comment by ybiquitous** [comment]
I'm closing this issue since this error doesn't likely happen. For example, look at [this demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcID0AqABAZwC4E8A2MeAlgHZYC0URGAhgEYFopIA6JAJCTQG7kAWMIgHM+WeGgBsABgAOADwDcbNgDo+NALZ0ArgCchMXeUhkapQ2mBs0acuRk0D-QSLFoAjFNmLraDTX1ScgIAMzcAFm8lEhsBYVFxMBo8MAAKbgDUuwcnONcASlsPL24+NABqNABiYE4eZ3isAF986JsZCFIsQ3IYbhgyDHESCBIYaKa2EAAaEBCiAgA5TThEGDlNGQIVDDAMDBnwUfmhBBArGLQWEHXukigMa-EAbV8ba+x8Qi7jY+FybA0e4BKAAvaPEC+AC6bCahxMJwAYhBdP4sGcAFYYUaHWAyA6IC7vECfAjEMhPK4gPA0brYa7TXwfXBkn4I-6A4G6UG7faU640ulYa6wkBNIA).

Please reopen it if you still have a problem.
